include/seqan3/alphabet/nucleotide/rna4.hpp

```diff
@@ -0,0 +1,263 @@
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file alphabet/nucleotide/rna4.hpp
+ * \ingroup alphabet
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Contains seqan3::rna4, container aliases and string literals.
+ */
+
+#pragma once
+
+#include <cassert>
+
+#include <string>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+
+// ------------------------------------------------------------------
+// rna4
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+/*!\brief The four letter RNA alphabet of A,C,G,U.
+ * \ingroup alphabet
+ *
+ * \details
+ * This alphabet inherits from seqan3::dna4 and is guaranteed to have the same internal representation of
+ * data. The only difference is that it prints 'U' on character conversion instead of 'T'. You assign
+ * between values of seqan3::dna4 and seqan3::rna4.
+ *
+ *~~~~~~~~~~~~~~~{.cpp}
+ *     rna4 my_letter{rna4::A};
+ *     // doesn't work:
+ *     // rna4 my_letter{'A'};
+ *
+ *     my_letter.assign_char('C'); // <- this does!
+ *
+ *     my_letter.assign_char('F'); // converted to A internally
+ *     if (my_letter.to_char() == 'A')
+ *        std::cout << "yeah\n"; // "yeah";
+ *~~~~~~~~~~~~~~~
+ */
+
+struct rna4 : public dna4
+{
+    /*!\name Letter values
+     * \brief Static member "letters" that can be assigned to the alphabet or used in aggregate initialization.
+     * \details Similar to an Enum interface . *Don't worry about the `internal_type`.*
+     */
+    //!\{
+    static const rna4 A;
+    static const rna4 C;
+    static const rna4 G;
+    static const rna4 U;
+    static const rna4 T;
+    static const rna4 UNKNOWN;
+    //!\}
+
+    /*!\name Read functions
+     * \{
+     */
+    //!\brief Return the letter as a character of char_type.
+    constexpr char_type to_char() const noexcept
+    {
+        return value_to_char[static_cast<rank_type>(_value)];
+    }
+    //!\}
+
+    /*!\name Write functions
+     * \{
+     */
+    //!\brief Assign from a character.
+    constexpr rna4 & assign_char(char_type const c) noexcept
+    {
+        _value = char_to_value[c];
+        return *this;
+    }
+
+    //!\brief Assign from a numeric value.
+    constexpr rna4 & assign_rank(rank_type const c)
+    {
+        assert(c < value_size);
+        _value = static_cast<internal_type>(c);
+        return *this;
+    }
+
+    //!\brief Assign from seqan3::dna4.
+    constexpr rna4 & operator=(dna4 const in) noexcept
```

Comment:
> why is this allowed but not the other way around?

The other way around already works automatically, because rna4 inherits dna4.

> no const ref?

dna4 is smaller than `&`